### PR TITLE
feat: [del-4488]: creating multiarch image for delegate

### PR DESCRIPTION
**Pull request by bharatgoelharness**
creating the multiarch image for delegate with no-tools

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/35470)

### CODE REVIEW

**Review by mdmilic — COMMENTED** [review]
Looks good, but can you append some suffix to the dockerfile name, multiarch or something

> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35470#-:-N7LCEzoBXkr0JxQX_mg:b-exsa8c)* status: 0 of 1 approvals obtained.

**Review by bharatgoelharness — COMMENTED** [review]
In future we will deprecate other file thats why we don't need to have targeted name. 

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/35470#-:-N7PEwZN6AOYJoOcbxD0:bxpovvq)* status: 1 file(s) to review. 0 of 1 approvals obtained.

File: dockerization/delegate/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+WORKDIR /opt/harness-delegate
+
+RUN chmod -R 755 /opt/harness-delegate
+
+ARG disable_client_tools
+
+ENV CLIENT_TOOLS_DOWNLOAD_DISABLED=$disable_client_tools
+
+RUN if [ "$TARGETPLATFORM" = "linux/arm64" ]; then curl -s https://app.harness.io/public/shared/jre/openjdk-11.0.14_9/jdk-11.0.14_linux-aarch64_bin.tar.gz | tar -xz; mkdir jdk-11.0.14+9-jre; mv jdk-11.0.14/* jdk-11.0.14+9-jre; rm -rf jdk-11.0.14; elif [ "$TARGETPLATFORM" = "linux/amd64" ]; then curl -s https://app.harness.io/public/shared/jre/openjdk-11.0.14_9/OpenJDK11U-jre_x64_linux_hotspot_11.0.14_9.tar.gz | tar -xz; fi
```
Comment (gaurav-jain1) [inline dockerization/delegate/Dockerfile:29]: let's throw an error if architecture doesn't match
if [ "$TARGETPLATFORM" = "linux/arm64" ]
  ...
elif [ "$TARGETPLATFORM" = "linux/amd64" ]
 ...
 else
   echo "Unsupported architecture type"
   exit
  fi

Comment (bharatgoelharness) [reply dockerization/delegate/Dockerfile]: done

**Comment by bharatgoelharness** [comment]
trigger codebasehash

**Pull request merged**